src/Components/CreateImageWizard/steps/Details/index.tsx

```diff
@@ -24,13 +27,32 @@ const DetailsStep = () => {
   const dispatch = useAppDispatch();
   const blueprintName = useAppSelector(selectBlueprintName);
   const blueprintDescription = useAppSelector(selectBlueprintDescription);
+  const distribution = useAppSelector(selectDistribution);
+  const arch = useAppSelector(selectArchitecture);
+  const targetEnvironments = useAppSelector(selectImageTypes);
+
+  useEffect(() => {
+    if (!blueprintName) {
+      dispatch(changeBlueprintName(generateDefaultName()));
+    }
+  }, [dispatch, blueprintName]);
   const handleNameChange = (
     _event: React.FormEvent<HTMLInputElement>,
     name: string
   ) => {
     dispatch(changeBlueprintName(name));
   };
 
+  const generateDefaultName = () => {
```

Comment:
I think this could be a function in a helper file and take `distribution`, `arch` `environments` as arguments :)
It's not a big deal, but it would be slightly cleaner.

Reply:
good idea